src/modules/flight_mode_manager/tasks/AutoFollowTarget/FlightTaskAutoFollowTarget.cpp

```diff
@@ -63,186 +66,243 @@ bool FlightTaskAutoFollowTarget::activate(const vehicle_local_position_setpoint_
 {
 	bool ret = FlightTask::activate(last_setpoint);
 
-	_position_setpoint = _position;
-
 	if (!PX4_ISFINITE(_position_setpoint(0)) || !PX4_ISFINITE(_position_setpoint(1))
 	    || !PX4_ISFINITE(_position_setpoint(2))) {
 		_position_setpoint = _position;
 	}
 
-	_target_position_filter.reset(Vector3f{NAN, NAN, NAN});
-	_offset_vector_filter.reset(Vector2f(0, 0));
-	_follow_angle_filter.reset(0.0f);
-	_velocity_ff_scale.reset(0.0f);
+	_target_pose_filter.reset(Vector3f{NAN, NAN, NAN});
+	_target_pose_filter.setParameters(TARGET_POSE_FILTER_NATURAL_FREQUENCY, TARGET_POSE_FILTER_DAMPING_RATIO);
 
-	// Initialize to something such that the drone at least points at the target, even if it's the wrong angle for the perspective.
-	// The drone will move into position as soon as the target starts moving and its heading becomes known.
-	Vector2f current_drone_heading_2d{cosf(_yaw), -sinf(_yaw)};
+	// We don't command the yawspeed, since only the yaw can be calculated off of the target
+	_yawspeed_setpoint = 0.f;
+	_yaw_setpoint_filter.reset(NAN);
 
-	if (PX4_ISFINITE(current_drone_heading_2d(0)) && PX4_ISFINITE(current_drone_heading_2d(1))) {
-		_offset_vector_filter.reset(current_drone_heading_2d);
+	// Update the internally tracaked Follow Target characteristics
+	_follow_angle_rad = math::radians(get_follow_me_angle_setting(_param_nav_ft_fs.get()));
+	_follow_target_distance = _param_nav_ft_dst.get();
+	_follow_target_height = _param_nav_ft_ht.get();
```

Comment:
Nope, I need to add this. Only thing that's bugging me was that we need to check if the parameter value has been updated, to not accidentally overwrite the RC adjusted value, if there is no Parameter Change. And to detect this 'update', we either :
1. Need to store local copy, and compare the new param value every loop
2. Override the "handleParamterUpdate()" function from Flight Task, to update the internal values accordingly, since this will be called when there's a parameter update system-wide.

However, the no.2 solution has a problem where a change in e.g. EKF2 parameter set, will also trigger the function. Not just about the Follow Target paramters.

I believe the best solution will be to add a function in the Params Class to check if it has been 'updated', and call this for each parameter to check for updates, but it hasn't been implemented in the Params class yet.

what do you think of these solutions?